R/adjacent.R

```diff
@@ -241,15 +241,15 @@ adjacent <- function(classimg, xpts = 100, xscale = NULL, bkgID = NULL,
         attr(classimg, "outline") <- polygon
       }
       # Multi images
-    } else if (multi_image) {
+    } else {
       if (length(polygon) != length(classimg)) {
         stop("The list of polygons must be equal in number to the list of images.")
       }
       if (!all(c("x", "y") %in% names(unlist(polygon)))) {
         message("Cannot find columns named x and y in outline, taking the first two columns as x-y coordinates")
-        polygon <- lapply(1:length(polygon), function(x) data.frame(x = polygon[[x]][, 1], y = polygon[[x]][, 2]))
+        polygon <- lapply(polygon, function(x) data.frame(x = x[, 1], y = x[, 2]))
       }
-      if (all(unlist(lapply(1:length(classimg), function(x) is.na(attr(classimg, "outline")))))) {
+      if (all(unlist(lapply(classimg, function(x) is.na(attr(x, "outline")))))) {
```

Comment:
I'm assuming that's what you meant here. You didn't use `x` in the original code.